test/unit/common/CommonUtils.test.ts

```diff
@@ -300,4 +362,26 @@ describe('CommonUtils', () => {
             parentPath: fname
         };
     }
+
+    async function verifyDownloadFails(url: string, mockResponse: any) {
+        if (url.startsWith('https:')) {
+            httpsGetStub.callsFake((_, callback) => {
+                setTimeout(() => callback(mockResponse), 100);
+                return { on: () => {}, end: () => {} };
+            });
```

Comment:
nit: this callback can be refactored outside of `if` since it's also used below in `httpsGetStub.callsFake(...)`.